diff -Naur a/evdev.inc b/evdev.inc
--- a/evdev.inc	2021-12-13 01:00:00.000000000 +0100
+++ b/evdev.inc	2022-08-04 17:19:16.965006368 +0200
@@ -431,6 +431,7 @@
   [0x01c1] = "KEY_DEL_EOS",
   [0x01c2] = "KEY_INS_LINE",
   [0x01c3] = "KEY_DEL_LINE",
+  [0x01c9] = "KEY_ASSISTANT",
   [0x01d0] = "KEY_FN",
   [0x01d1] = "KEY_FN_ESC",
   [0x01d2] = "KEY_FN_F1",
diff -Naur a/event-input.c b/event-input.c
--- a/event-input.c	2021-12-13 01:00:00.000000000 +0100
+++ b/event-input.c	2022-08-04 17:42:58.341320016 +0200
@@ -78,6 +78,11 @@
 # define KEY_CAMERA_FOCUS               0x0210
 #endif

+#ifndef KEY_ASSISTANT
+/** Xperia 10 III Assistant button */
+#define KEY_ASSISTANT                   0x01c9
+#endif
+
 #ifndef FF_STATUS_CNT
 # ifdef FF_STATUS_MAX
 #  define FF_STATUS_CNT (FF_STATUS_MAX+1)
@@ -1279,6 +1284,7 @@
         /* Home key should be handled by mce and can be
          * ignored as well. */
         KEY_HOME,
+        KEY_ASSISTANT,
         -1
     };

diff -Naur a/powerkey.c b/powerkey.c
--- a/powerkey.c	2021-12-13 01:00:00.000000000 +0100
+++ b/powerkey.c	2022-08-04 17:42:56.665309287 +0200
@@ -100,6 +100,12 @@

 #define MODULE_NAME "powerkey"

+#ifndef KEY_ASSISTANT
+/** Xperia 10 III Assistant button */
+#define KEY_ASSISTANT                   0x01c9
+#endif
+
+
 /* ========================================================================= *
  * PROTOTYPES
  * ========================================================================= */
@@ -3255,6 +3261,18 @@
             }
             break;

+        // treat assistant button as HOME button (note the function call)
+        case KEY_ASSISTANT:
+            if( ev->value == 1 ) {
+                mce_log(LL_CRUCIAL, "assistant key pressed");
+                homekey_stm_set_pressed(true);
+            }
+            else if( ev->value == 0 ) {
+                mce_log(LL_CRUCIAL, "assistant key released");
+                homekey_stm_set_pressed(false);
+            }
+            break;
+
         default:
             break;
         }
